configure.ac

```diff
@@ -668,8 +668,20 @@ AC_PROG_LD_GNU
 
 AC_CHECK_HEADERS(sys/filio.h sys/sockio.h netdb.h utime.h sys/utime.h semaphore.h sys/un.h linux/rtc.h sys/syscall.h sys/mkdev.h sys/uio.h sys/param.h sys/sysctl.h libproc.h sys/prctl.h copyfile.h)
 AC_CHECK_HEADERS(sys/param.h sys/socket.h sys/ipc.h sys/utsname.h alloca.h ucontext.h pwd.h sys/select.h netinet/tcp.h netinet/in.h unistd.h sys/types.h link.h asm/sigcontext.h sys/inotify.h arpa/inet.h complex.h unwind.h)
-AC_CHECK_HEADER(unistd.h, [HAVE_UNISTD_H=1], [HAVE_UNISTD_H=0])
+
+# zlib/configure checks for stdarg.h existance and defines HAVE_STDARG_H on compiler
+# command line (not config.h) if it is present.
+CFLAGS="$CFLAGS -DHAVE_STDARG_H"
```

Comment:
Is there a reason you're not using `AC_CHECK_HEADER` here but are for unistd.h?

Reply:
zlib does not use config.h.

Reply:
The comment explains -- "compiler command line...not config.h"

Reply:
I'm confused because you put the same comment next to unistd.h and then used AC_CHECK_HEADER anyway, but I guess you were just preserving the existing behavior in our configure?

Reply:
Maybe I'm just misunderstanding something - I haven't dealt with autotools much.

Reply:
Oh, sorry. This is because I am assuming stdarg.h exists. Which mono already does in a few places. And should exist in any implementation since at least 1989. Maybe some day we port to K&R, the older varargs form?
zlib does not assume stdarg.h exists though.

unistd.h is at least Posix only, and sometimes mono assumes it exists, sometimes not.
There a few forms to unistd.h in mono which seems dubious.

Reply:
Maybe this is overkill though.

```
#if defined(STDC) || defined(Z_HAVE_STDARG_H)
#  ifndef Z_SOLO
#    include <stdarg.h>         /* for va_list */
#  endif
#endif

#if !defined(STDC) && (defined(__STDC__) || defined(__cplusplus))
#  define STDC
#endif
```

Reply:
And the only use is here:

```
#if defined(STDC) || defined(Z_HAVE_STDARG_H)
#  ifndef Z_SOLO
ZEXTERN int            ZEXPORTVA gzvprintf Z_ARG((gzFile file,
                                                  const char *format,
                                                  va_list va));
#  endif
#endif
```

which is unused and I deleted the file I think. So the configure.ac change might be doubly unneeded, and also _fairly_ harmless. I do like short compiler command lines though and this regresses that.